django/db/models/lookups.py

```diff
@@ -286,32 +208,34 @@ def get_rhs_op(self, connection, rhs):
 
     def as_sql(self, compiler, connection):
         max_in_list_size = connection.ops.max_in_list_size()
-        if self.rhs_is_direct_value() and (max_in_list_size and
-                                           len(self.rhs) > max_in_list_size):
-            # This is a special case for Oracle which limits the number of elements
-            # which can appear in an 'IN' clause.
-            lhs, lhs_params = self.process_lhs(compiler, connection)
-            rhs, rhs_params = self.batch_process_rhs(compiler, connection)
-            in_clause_elements = ['(']
-            params = []
-            for offset in range(0, len(rhs_params), max_in_list_size):
-                if offset > 0:
-                    in_clause_elements.append(' OR ')
-                in_clause_elements.append('%s IN (' % lhs)
-                params.extend(lhs_params)
-                sqls = rhs[offset: offset + max_in_list_size]
-                sqls_params = rhs_params[offset: offset + max_in_list_size]
-                param_group = ', '.join(sqls)
-                in_clause_elements.append(param_group)
-                in_clause_elements.append(')')
-                params.extend(sqls_params)
+        if self.rhs_is_direct_value() and max_in_list_size and len(self.rhs) > max_in_list_size:
+            return self.split_parameter_list_as_sql(compiler, connection)
+        return super(In, self).as_sql(compiler, connection)
+
+    def split_parameter_list_as_sql(self, compiler, connection):
+        # This is a special case for Oracle which limits the number of elements
```

Comment:
I guess we could remove the mention of Oracle and just say "for databases which limit..."